### PR TITLE
Adjust Screenshots tests to reflect latest UI changes(EXPOSUREAPP-4878)

### PR DESCRIPTION
**Pull request by mtwalli**
This PR is based on the changes made in #2235. Please check it first

- Adjusted the following screenshots tests:
  1- Home screen tests
  2- Contact diary tests
  3- Statistics tests

- Refactor Screenshot logic
- Update versions in  Gemfile.lock 

## Testing: Same as before for screenshots testing
1- Build APKs required for Screengrab
2- run fastlane screengrab

### CODE REVIEW

File: Corona-Warn-App/src/main/AndroidManifest.xml
Comment (d4rken) [inline Corona-Warn-App/src/main/AndroidManifest.xml:79]: So this PR needs to be merged after the TABBar PR, right?

**Review by d4rken — COMMENTED** [review]
Need to re-review this once the TabBar PR is merged.

File: Corona-Warn-App/build.gradle
Comment (mtwalli) [reply Corona-Warn-App/build.gradle:382]: it is needed for ` launchAcitivity`

File: Corona-Warn-App/src/androidTest/java/de/rki/coronawarnapp/ui/main/MainActivityTest.kt
Comment (mtwalli) [reply Corona-Warn-App/src/androidTest/java/de/rki/coronawarnapp/ui/main/MainActivityTest.kt:78]: We can ,but I don't think it make sense. It is screenshots tests it should reflect what is in the App. When the home fragment is displayed the bottom bar should be present right?

File: Corona-Warn-App/src/main/AndroidManifest.xml
Comment (mtwalli) [reply Corona-Warn-App/src/main/AndroidManifest.xml:79]: Yes

File: Corona-Warn-App/src/androidTest/java/de/rki/coronawarnapp/ui/main/MainActivityTest.kt
Comment (d4rken) [reply Corona-Warn-App/src/androidTest/java/de/rki/coronawarnapp/ui/main/MainActivityTest.kt:78]: Ok

File: Corona-Warn-App/src/androidTest/java/de/rki/coronawarnapp/ui/main/MainActivityTest.kt
Comment (jurajkusnier) [inline Corona-Warn-App/src/androidTest/java/de/rki/coronawarnapp/ui/main/MainActivityTest.kt:277]: I guess you need to keep default delay here, the first and the second screenshots look the same

**Comment by harambasicluka** [comment]
I added the label `author merge` due to this comment: https://github.com/corona-warn-app/cwa-app-android/pull/2247#discussion_r568430174
This PR has to be merged first: #2235

File: Corona-Warn-App/src/androidTest/java/de/rki/coronawarnapp/ui/contactdiary/ContactDiaryOverviewFragmentTest.kt
Comment (mtwalli) [reply Corona-Warn-App/src/androidTest/java/de/rki/coronawarnapp/ui/contactdiary/ContactDiaryOverviewFragmentTest.kt:85]: this is moved to MainAcitivtyTest

File: Corona-Warn-App/src/androidTest/java/de/rki/coronawarnapp/ui/onboarding/OnboardingFragmentTest.kt
Comment (jurajkusnier) [inline Corona-Warn-App/src/androidTest/java/de/rki/coronawarnapp/ui/onboarding/OnboardingFragmentTest.kt:59]: This also needs default delay otherwise screenshots look the same

Comment (mtwalli) [reply Corona-Warn-App/src/androidTest/java/de/rki/coronawarnapp/ui/onboarding/OnboardingFragmentTest.kt:59]: I don't think so, there was no time delay before here. Look into the changed code

Comment (mtwalli) [reply Corona-Warn-App/src/androidTest/java/de/rki/coronawarnapp/ui/onboarding/OnboardingFragmentTest.kt:59]: Delay was introduced just to give espresso the time to display the layout. in some cases the layout is already displayed and you need to perform an action

Comment (jurajkusnier) [reply Corona-Warn-App/src/androidTest/java/de/rki/coronawarnapp/ui/onboarding/OnboardingFragmentTest.kt:59]: image
in that case it didn't work even before the update

Comment (mtwalli) [reply Corona-Warn-App/src/androidTest/java/de/rki/coronawarnapp/ui/onboarding/OnboardingFragmentTest.kt:59]: does it work with the delay?

**Review by jurajkusnier — APPROVED** [review]
LGTM

**Review by ralfgehrer — APPROVED** [review]
LGTM. I did NOT run all screenshots - please wait for Matthias' approval. He is currently executing the entire set.

**Pull request merged**
